kernel/kernel-defaults/src/test/scala/io/delta/kernel/defaults/internal/expressions/DefaultExpressionEvaluatorSuite.scala

```diff
@@ -307,6 +307,167 @@ class DefaultExpressionEvaluatorSuite extends AnyFunSuite with ExpressionSuiteBa
       "Coalesce is only supported for boolean type expressions")
   }
 
+  test("evaluate expression: like") {
+    val col1 = stringVector(Seq[String](
+      "one", "two", "three", "four", null, null, "seven", "eight"))
+    val col2 = stringVector(Seq[String](
+      "one", "Two", "thr%", "four%", "f", null, null, "%ght"))
+    val schema = new StructType()
+      .add("col1", StringType.STRING)
+      .add("col2", StringType.STRING)
+    val input = new DefaultColumnarBatch(col1.getSize, schema, Array(col1, col2))
+
+    def checkBatch(
+          input: DefaultColumnarBatch,
+          likeExpression: Predicate,
+          expOutputSeq: Seq[BooleanJ]): Unit = {
+      val actOutputVector =
+        new DefaultExpressionEvaluator(
+          schema, likeExpression, BooleanType.BOOLEAN).eval(input)
+      val expOutputVector = booleanVector(expOutputSeq);
+      checkBooleanVectors(actOutputVector, expOutputVector)
+    }
+
+    // check column expressions on both sides
+    checkBatch(
+      input, like(new Column("col1"), new Column("col2")),
+        Seq[BooleanJ](true, false, true, true, null, null, null, true))
+
+    // check column expression against literal
+    checkBatch(
+      input, like(new Column("col1"), Literal.ofString("t%")),
+        Seq[BooleanJ](false, true, true, false, null, null, false, false))
+
+    // ends with checks
+    checkBatch(
+      input, like(new Column("col1"), Literal.ofString("%t")),
+        Seq[BooleanJ](false, false, false, false, null, null, false, true))
+
+    // contains checks
+    checkBatch(
+      input, like(new Column("col1"), Literal.ofString("%t%")),
+        Seq[BooleanJ](false, true, true, false, null, null, false, true))
+
+    val dummyInput = new DefaultColumnarBatch(1,
+        new StructType().add("dummy", StringType.STRING),
+        Array(stringVector(Seq[String](""))))
+
+    def checkLiteral(left: String, right: String,
+        escape: Character = null, expOutput: BooleanJ): Unit = {
+      val expression = like(Literal.ofString(left), Literal.ofString(right), Option(escape))
+      checkBatch(dummyInput, expression, Seq[BooleanJ](expOutput))
+    }
+
+    Seq('!', '@', '#').foreach {
+      escape => {
+        // null/empty
+        checkLiteral(null, "a", null, null)
```

Comment:
nit: rename `checkLiteral` to `checkLike`. Makes it easy to understand that it is checking the `like` expression.